app/models/document/document_filter.rb

```diff
@@ -73,6 +73,7 @@ def to_date
   def unpaginated_documents
     documents = Document
     documents = documents.where(block_type: Schema.valid_schemas)
+    documents = documents.where(hide_from_search_index: false) unless Current.user&.is_e2e_user?
```

Comment:
If we don't currently have other scenarios for needing to hide blocks, is it clearer to have just one term in play. e.g. for our finder to say:

- "exclude documents made in testing unless I'm the tester" rather than
- "exclude documents marked as hidden unless I'm the tester"